### DISCUSSION TITLE
Executor module

### DISCUSSION THREAD

**Discussion by bhavaniravi** [bhavaniravi/kubetask#6] [Ideas]
> The module that spins off a kubernetes pod for each task

1. Read the task details from the queue
2. Create a kubernetes pod
3. Start the pod
4. Poll the pod or run a side-car(callback) to mark the end of the task
5. Once the task is complete update the respective task and task instance
6. Collect the logs and push it to file storage/S3 bucket
7. we should be able to stream the logs to the UI when the task is running

**Comment by Catastrophe0123** [comment]
i have some questions:
what are you planning on the format of the config? is it gonna be an amalgamation of airflow and celery or somethign like a docker config?
in the docs, the worker module spec reads more like a scheduler. i think something like a master or TaskDaemon which takes care of the following points: 
* Read the task details from the queue
* Create a kubernetes pod
* Start the pod
* Poll the pod or run a side-car(callback) to mark the end of the task
* Once the task is complete update the respective task and task instance

would abstract away the worker logic and reduces it to just running the task and pushing to blobstore

**Comment by bhavaniravi** [comment]
You're right It's more of a `TaskExecutor` or `Orchestrator` than a worker. Renamed the spec accordingly. Will create another one for worker
